specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2020-06-25/guestconfiguration.json

```diff
@@ -0,0 +1,1317 @@
+        },
+        "endTime": {
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true,
+          "description": "End date and time of the guest configuration assignment compliance status check."
+        },
+        "complianceStatus": {
+          "type": "string",
+          "readOnly": true,
+          "description": "A value indicating compliance status of the machine for the assigned guest configuration.",
+          "x-ms-enum": {
+            "name": "complianceStatus",
+            "modelAsString": true
+          },
+          "enum": [
+            "Compliant",
+            "NonCompliant",
+            "Pending"
+          ]
+        },
+        "operationType": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Type of report, Consistency or Initial",
+          "x-ms-enum": {
+            "name": "type",
+            "modelAsString": true
+          },
+          "enum": [
+            "Consistency",
+            "Initial"
+          ]
+        },
+        "resources": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AssignmentReportResourceMetadata"
+          },
+          "description": "The list of resources for which guest configuration assignment compliance is checked."
+        }
+      }
+    },
+    "GuestConfigurationAssignmentProperties": {
+      "properties": {
+        "targetResourceId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "VM resource Id.",
+          "x-nullable": true
+        },
+        "subscriptionId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Subscription Id to which the Guest Configuration assignment resource is assigned to.",
+          "x-nullable": true
+        },
+        "guestConfiguration": {
+          "$ref": "#/definitions/GuestConfigurationNavigation",
+          "description": "The guest configuration to assign."
+        },
+        "complianceStatus": {
+          "type": "string",
+          "readOnly": true,
+          "description": "A value indicating compliance status of the machine for the assigned guest configuration.",
+          "x-ms-enum": {
+            "name": "complianceStatus",
+            "modelAsString": true
+          },
+          "enum": [
+            "Compliant",
+            "NonCompliant",
+            "Pending"
+          ]
+        },
+        "lastComplianceStatusChecked": {
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true,
+          "description": "Date and time when last compliance status was checked.",
+          "x-nullable": true
+        },
+        "latestReportId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Id of the latest report for the guest configuration assignment. ",
+          "x-nullable": true
+        },
+        "latestAssignmentReport": {
```

Comment:
>latestAssignmentReport [](start = 9, length = 22)

pulling the content of children into a parent is something that should be avoided unless absolutely necessary.  It breaks RBAC since users without /read permissions on /reports will now be able to see the report in the guestConfigAssignment

Reply:
these new properties are required to provide a complete data set for a guest configuration assignment resource. this would be sent to ARG and for guest config customers/partners to query. I do understand your concern wrt RBAC but i cant think of a scenario where a user would have access to guest config resource but not its child resource - report in this case. As all the relevant compliance information is in the report.
having said that - i am open to other suggestions that you may have to achieve the same.

Reply:
I was thinking about the RBAC thing more - and in the arm manifest we only list one proxy type which is guestConfigurationAssignment type. We dont specify reports as a child type. so i am wondering how would someone be able to place RBAC rules on the reports when its not actually a type? Am i missing something here?


"resourceTypes": [
        {
            "name": "guestConfigurationAssignments",
            "routingType": "ProxyOnly, Extension",
            "resourceDeletionPolicy": "Cascade",
            "linkedOperationRules": [],
            "quotaRule": {
                "quotaPolicy": "None"
            }

Reply:
RBAC is based on your operations API:

Operation         : Microsoft.GuestConfiguration/guestConfigurationAssignments/reports/read
OperationName     : Microsoft.GuestConfiguration/guestConfigurationAssignments/reports/read
ProviderNamespace : Microsoft Guest Configuration
ResourceName      : Microsoft.GuestConfiguration/guestConfigurationAssignments
Description       : Get guest configuration assignment report.
IsDataAction      : False

Reply:
Since its all readonly and not modifiable at either level I won't block on it but this is pretty messy just to get ARG queries to work